### PR TITLE
[vulkan] Add Winograd convolutions

### PR DESCRIPTION
**Pull request by SS-JIA**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#54639 [vulkan] Add Winograd convolutions**

Differential Revision: [D27514882](https://our.internmc.facebook.com/intern/diff/D27514882)

### CODE REVIEW

File: aten/src/ATen/test/vulkan_api_test.cpp [hunk 1]
```diff
@@ -606,6 +606,78 @@ TEST(VulkanAPITest, conv2d_pw) {
...
+        output_channels,
+        input_channels,
+        width,
+        height,
+      };
+    }
+  } weights {13, input.channels, 3, 3};
+
+  const auto input_cpu = at::rand(input.size(), at::device(at::kCPU).dtype(at::kFloat));
+  const auto weights_cpu = at::rand(weights.size(), at::device(at::kCPU).dtype(at::kFloat))*2;
```
Comment (IvanKobzarev) [inline aten/src/ATen/test/vulkan_api_test.cpp:652]: why '*2' in the end?

File: aten/src/ATen/test/vulkan_api_test.cpp [hunk 2]
```diff
@@ -430,11 +430,11 @@ TEST(VulkanAPITest, conv2d) {
...
     }
-  } weights {83, input.channels, 13, 2};
+  } weights {1, input.channels, 3, 3};
 
-  const auto input_cpu = at::randn(input.size(), at::device(at::kCPU).dtype(at::kFloat));
-  const auto weights_cpu = at::randn(weights.size(), at::device(at::kCPU).dtype(at::kFloat));
-  const auto bias_cpu = at::randn({weights.output_channels}, at::device(at::kCPU).dtype(at::kFloat));
+  const auto input_cpu = at::randn(input.size(), at::device(at::kCPU).dtype(at::kFloat))*5;
+  const auto weights_cpu = at::randn(weights.size(), at::device(at::kCPU).dtype(at::kFloat))*5;
+  const auto bias_cpu = at::randn({weights.output_channels}, at::device(at::kCPU).dtype(at::kFloat))*5;
```
Comment (IvanKobzarev) [inline aten/src/ATen/test/vulkan_api_test.cpp:437]: why *5, testing ?

Comment (SS-JIA) [reply aten/src/ATen/test/vulkan_api_test.cpp:437]: ah, I added these during testing and forgot to remove them, same for the one below. Good eye!

File: aten/src/ATen/test/vulkan_api_test.cpp (hunk 1 above)
Comment (SS-JIA) [reply aten/src/ATen/test/vulkan_api_test.cpp:652]: Same as above

**Review by IvanKobzarev — APPROVED** [review]

**Pull request closed without merging**